client/state/preferences/test/actions.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import nock from 'nock';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_RECEIVE,
+	PREFERENCES_FETCH,
+	PREFERENCES_FETCH_SUCCESS,
+	PREFERENCES_SET
+} from 'state/action-types';
+
+import { useSandbox } from 'test/helpers/use-sinon';
+import { DEFAULT_PREFERENCES, USER_SETTING_KEY } from '../constants';
+import { fetchPreferences, setPreference } from '../actions';
+
+describe( 'actions', () => {
+	let sandbox, spy;
+
+	useSandbox( newSandbox => {
+		sandbox = newSandbox;
+		spy = sandbox.spy();
+	} );
+	const responseShape = {
+		_headers: { 'content-type': 'application/json' },
+		[ USER_SETTING_KEY ]: DEFAULT_PREFERENCES
+	};
+	const getState = () => ( { preferences: { values: DEFAULT_PREFERENCES } } );
+	before( () => {
+		nock( 'https://public-api.wordpress.com:443' )
+			.persist()
+			.get( '/rest/v1.1/me/settings' )
+			.reply( 200, responseShape )
+			.post( '/rest/v1.1/me/settings/' )
+			.reply( 200, responseShape );
```

Comment:
I personally find using `nock` as tricky, too much setup required and it's also super brittle. One small change in `wpcom` can break so many tests at once... 

Another approach to try in the future would be to stub `wpcom.me().settings().get()` to return resolved Promise with data only. I did something similar in another project. I hope it's also possible with wpcom :D